core/views/MemberViews.py

```diff
@@ -41,12 +67,21 @@ def post(self, request, *args, **kwargs):
         return self.get(request, *args, **kwargs)
 
 
-class MemberFinishView(UpdateView):
+class MemberFinishView(UserPassesTestMixin, UpdateView):
 
     model = Member
     form_class = MemberFinishForm
     template_name_suffix = "_finish"
 
+    raise_exception = True
+    permission_denied_message = "You are not allowed complete the sign up process for anyone but yourself!"
+
+    def test_func(self):
```

Comment:
I think Pytest can be confused if you name other methods than actual unit tests test

Reply:
I can't unfortunately, test_func is a name enforced by django